### ISSUE TITLE
goal: Jan Mobile MVP

### ISSUE THREAD

**Issue by dan-menlo** [menloresearch/jan#6489]
## 🎯 Goal
Develop a functional Jan mobile application MVP for Android and iOS platforms with core chat functionality.

## 📖 Context
* Jan is currently available exclusively on desktop platforms (macOS, Windows, Linux) and web browsers
* Mobile platform availability is required to enhance competitive positioning against established chat applications, strengthen brand identity, and expand user reach
* Mobile deployment represents a critical milestone for market expansion and user accessibility

## ✅ Scope
* Development and deployment of native Android and iOS applications
* Implementation of configurable remote API model integration capabilities
  * Able to connect to Jan Local API Server
* Mobile-optimized user interface design and responsive layout implementation

## ❌ Out of Scope
* Local model download and execution on mobile devices

## 🛠 Deliverables
* Beta builds for Android and iOS platforms suitable for internal testing and user feedback
* Documentation covering manual installation procedures and configuration

## 📊 Success Criteria
* Beta applications successfully deployed on Android and iOS devices 
* Mobile applications demonstrate full functional parity with desktop version for remote API model integration
* User interface passes usability testing across various mobile device form factors (phones and tablets)
* Remote API connectivity maintains stable performance

**Comment by ux-han** [comment]
Thanks @urmauur. Here are 2 design options for mobile:

# Design 
## Option 1
Figma link: https://www.figma.com/design/HI8LuAGRsYQ2FdTRtg3aHH/Jan-Mobile?node-id=0-1

- Keep threads as base / fundamental screen
- Bundle Hub & Settings as overlay functions

![Screenshot 2024-10-31 at 8 40 41 AM](https://github.com/user-attachments/assets/16a0a859-ab99-48a9-8a72-815d026272ca)
![Screenshot 2024-10-31 at 8 40 53 AM](https://github.com/user-attachments/assets/335665ca-ace0-48d2-b523-f00e4782e3a9)
![Screenshot 2024-10-31 at 8 40 58 AM](https://github.com/user-attachments/assets/0dcba627-bcf2-4f02-931c-238f47ba459e)


## Option 2
Figma link: https://www.figma.com/design/HI8LuAGRsYQ2FdTRtg3aHH/Jan-Mobile?node-id=26-1549

Based on @urmauur's suggestion & inspired by Obsidian

- Keep Ribbon as bottom main navigation
- However, we’ll need to adapt some elements for mobile to ensure a user-friendly experience. Unlike on desktop, where the right panel can remain open alongside the chat, mobile screens are too limited to support this without compromising usability.
- For example, the model selector on mobile should ideally be placed at the top of the chat screen. This allows users to easily switch models without needing to toggle the right panel open and closed repeatedly. It also helps reduce clutter in the bottom navigation and keeps the input box focused on messaging, rather than adding additional controls.

![Screenshot 2024-10-31 at 8 45 02 AM](https://github.com/user-attachments/assets/b8d92851-4dda-436b-a7c6-5799ee1c406a)
![Screenshot 2024-10-31 at 8 45 36 AM](https://github.com/user-attachments/assets/79acdb56-d257-46b7-8a6a-cc3ce90e672c)
![Screenshot 2024-10-31 at 8 45 46 AM](https://github.com/user-attachments/assets/5ef63990-2f9c-4d19-acd9-741dbdc71c33)


# Set up models to use on mobile
It's clear that users can't run large models on mobile. Here are some options for them:
**1. Use models via nearby servers:**
- Via IP address: If users choose to connect to a server on the same network, the server could generate a QR code with connection details (e.g., IP address, port). Users can simply scan the code with Jan on their mobile device to initiate the connection or enter the IP address manually. 
- Via Bluetooth device pairing:  For users without Wi-Fi access, we can implement Bluetooth-based pairing, making it straightforward to connect devices without entering complex information.
2. Recommend tiny on-device models to download
3. Connect cloud services

![Screenshot 2024-10-31 at 8 51 05 AM](https://github.com/user-attachments/assets/8b84007e-0a1c-4fd9-854c-9b98b9ed78e5)

**Comment by urmauur** [comment]
Since we allow the community to inject icons from extensions, let’s say I install 10 extensions, and all of them create new icons on the ribbon. How would this menu look? 

If we go with this approach, we need to handle this case.

<img width="643" alt="Screenshot 2024-11-01 at 10 11 20" src="https://github.com/user-attachments/assets/1287344f-ac50-4bf0-bcc1-18c7926f4b36">

**Comment by louis-jan** [comment]
UPDATE:
- Need to create Tauri target for mobile.
- Can spawn the web version of app to unblock the development process.
